pkg/relayer/proxy/interface.go

```diff
@@ -0,0 +1,26 @@
+package proxy
+
+import (
+	"context"
+
+	"pocket/pkg/observable"
+	"pocket/x/service/types"
+)
+
+// RelayerProxy is the interface for the proxy that serves relays to the application.
+// It is responsible for starting and stopping all supported proxies.
+// While handling requests and responding in a closed loop, it also notifies
+// the miner about the relays that have been served.
+type RelayerProxy interface {
+
+	// Start starts all supported proxies and returns an error if any of them fail to start.
+	Start(ctx context.Context) error
+
+	// Stop stops all supported proxies and returns an error if any of them fail.
+	Stop() error
+
+	// ServedRelays returns an observable that notifies the miner about the relays that have been served.
+	// A served relay is one whose RelayRequest's signature and session have been verified,
+	// and its RelayResponse has been signed and successfully sent to the client.
+	ServedRelays() observable.Observable[*types.Relay]
```

Comment:
When we get to the implementation, I want to see how this is used and might ask to rename it to `RelayServer` since it does not serve/return static data.

Not a blocker for this PR (merge as is), but will follow up.

Reply:
Basically `RelayerProxy.ServedRelays` would be ranged over from the miner to update the corresponding `SMST`.

It would be very similar to what we have in the alpha repo (simplified pseudo go version):
```go
// goHandleRelays is called as a go routine when the `Miner` starts
func (m *Miner) goHandleRelays() {
  ch := m.relayerProxy.ServedRelays().Subscribe().Ch()
  for relay := range ch {
    go m.goHandleSingleRelay(relay)
  }
}

func (m *Miner) goHandleSingleRelay(relay *types.Relay) {
    smst := m.getTreeForSession()
    smst.Update(relay.Hash(), relay.Marshal(), 1)
}
```

Will document this in the next PR.

Reply:
This explanation makes me think even more that we should rename it.

You don't subscribe to `ServedRelays`, but rather to a `RelayServer` or `RelayObservable` or other.

@bryanchriswhite  wdyt?